crates/biome_js_analyze/src/aria_analyzers/a11y/no_aria_unsupported_elements.rs

```diff
@@ -120,4 +122,27 @@ impl Rule for NoAriaUnsupportedElements {
             }),
         )
     }
+
+    fn action(_ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let element = _ctx.query();
+        let mut mutation = _ctx.root().begin();
+        let mut removed_attribute = "role".to_string();
+
+        for attribute in element.attributes() {
+            let attribute = attribute.as_jsx_attribute()?;
+            let attribute_name = attribute.name().ok()?.as_jsx_name()?.value_token().ok()?;
+            let attribute_name = attribute_name.to_string();
+            removed_attribute = attribute_name;
+            mutation.remove_node(attribute.clone());
+        }
```

Comment:
I missed the fact that you remove every attribute of the elements. Some attribute can be valid (for example in the snapshot, the `charset` attribute should not be removed`.

Instead, we could search for the attribute using `find_map`. SOmething like this:

```suggestion
        let attribute = element.attributes().into_iter().find_map(|attribute| {
            let jsx_attribute = attribute.as_jsx_attribute()?;
            let attribute_name = jsx_attribute.name().ok()?.as_jsx_name()?.value_token().ok()?;
            let attribute_name = attribute_name.text_trimmed();
            (attribute_name.starts_with("aria-") || attribute_name == "role").then_some(attribute)
        })?;
        // remove attribute and extarct its name
```

Another alternative is replacing the rule state by the attribute to remove.

Reply:
Oh, thanks I thought that it only had valid aria and role attributes in the list and did not consider others.